libs/net/template/ReportEngine.cs

```diff
@@ -0,0 +1,214 @@
+
+using System.Text;
+using System.Text.Json;
+using Microsoft.Extensions.Options;
+using TNO.Core.Extensions;
+using TNO.Core.Http;
+using TNO.TemplateEngine.Config;
+using TNO.TemplateEngine.Models.Charts;
+using TNO.TemplateEngine.Models.Reports;
+
+namespace TNO.TemplateEngine;
+
+/// <summary>
+/// ReportEngine class, provides a centralize collection of methods to generate reports and charts.
+/// </summary>
+public class ReportEngine : IReportEngine
```

Comment:
Refactored a lot of code to move it here.  This way we can edit one function to update/fix reports/charts.